nuhal_tiva/src/pin_tiva.c

```diff
@@ -200,7 +244,76 @@ bool pin_read(uint32_t pin)
     const uint32_t base = pin_base(pin);
     const uint32_t mask = pin_mask(pin);
     return HWREG(base + (GPIO_O_DATA + (mask << 2)));
+}
+
+void analog_pin_read(uint32_t* pins, size_t num_pins, bool use_adc_1, uint16_t* out)
+{
+    if (num_pins > 8)
+    {
+        error(FILE_LINE, "Too many pins to analog read from");
+    }
+
+    uint32_t bases[8] = {0};
+    uint32_t masks[8] = {0};
+    uint8_t ains[8] = {0};
+    uint32_t adc_raw[8] = {0};
+
+     // options for analog pin:
+    // - option 1: Using ADC0 or ADC1
+    // - option 2: Using SS0 (sample up to 8 times sequentially), SS1 (up to 4), SS2 (up to 4), OR SS3 (only 1)
+    // - option 3: How many pins are we reading from sequentially and from which ain channel?
+    
+    // Option 1: ADC0 or ADC1
+    uint32_t adcsysctl = use_adc_1 ? SYSCTL_PERIPH_ADC1 : SYSCTL_PERIPH_ADC0;
+    uint32_t adcbase   = use_adc_1 ? ADC1_BASE : ADC0_BASE;
 
+    // Option 2: find appropriate sample sequencer for number of pins that will be read from
+    uint8_t ss = 0;
+    if (num_pins > 4)
+    {
+        ss = 0; // SS0 supports up to 8 steps
+    }
+    else if (num_pins > 1)
+    {
+        ss = 1; // SS1 supports up to 4 steps
+    }
+    else
+    {
+        ss = 3; // SS3 supports 1 step
+    }
+
+    tiva_peripheral_enable(adcsysctl);
```

Comment:
We should not be enabling a perifpheral every time we want to read from the adc. This is part of the setup for the adc.